pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -401,15 +401,42 @@ func (service *SubnetPortService) AllocatePortFromSubnet(subnet *model.VpcSubnet
 
 	info.lock.Lock()
 	defer info.lock.Unlock()
+
 	if !ok {
-		totalIP := int(*subnet.Ipv4SubnetSize)
-		if len(subnet.IpAddresses) > 0 {
-			// totalIP will be overrided if IpAddresses are specified.
-			totalIP, _ = util.CalculateIPFromCIDRs(subnet.IpAddresses)
+		dhcpMode := "DHCP_DEACTIVATED"
+		subnetInfo, _ := servicecommon.ParseVPCResourcePath(*subnet.Path)
+		if subnet.SubnetDhcpConfig != nil && subnet.SubnetDhcpConfig.Mode != nil {
+			dhcpMode = *subnet.SubnetDhcpConfig.Mode
+		}
+		// For DHCP Deactivated mode Subnet, get total IPs from IP pool static-ipv4-default
+		if dhcpMode == "DHCP_DEACTIVATED" {
+			staticIPPool, err := service.NSXClient.IPPoolClient.Get(subnetInfo.OrgID, subnetInfo.ProjectID, subnetInfo.VPCID, subnetInfo.ID, "static-ipv4-default")
+			if err != nil {
```

Comment:
If you get the totalIP by using API, will you cache it by path so no need to call the API again?

Reply:
Yea, I thought about it, it only applies to static subnet. But how to remove the subnet from the cache?

for dhcp subnet, since user might update its reserved_ip, then dhcp subnet pool size might change, so need to get the pool size from NSX.